src/main/java/blackjack/domain/result/DealerResult.java

```diff
@@ -0,0 +1,42 @@
+package blackjack.domain.result;
+
+import blackjack.domain.common.Name;
+
+import java.util.EnumMap;
+import java.util.List;
+
+public class DealerResult {
+    private final Name name;
+    private final EnumMap<ResultStatus, Count> resultStatusBoard;
+
+    public DealerResult(final Name name, final EnumMap<ResultStatus, Count> resultStatusBoard) {
+        this.name = name;
+        this.resultStatusBoard = resultStatusBoard;
+    }
+
+    public static DealerResult of(final Name name, final List<GamePlayerResult> gamePlayerResults) {
+        final EnumMap<ResultStatus, Count> resultStatusBoard = new EnumMap(ResultStatus.class);
+
+        for (final GamePlayerResult gamePlayerResult : gamePlayerResults) {
+            increment(resultStatusBoard, gamePlayerResult.getResultStatus()
+                                                         .reverse());
+        }
+
+        return new DealerResult(name, resultStatusBoard);
```

Comment:
stream을 사용하면 아래처럼도 가능합니다
```suggestion
    Map<ResultStatus, Count> collect = gamePlayerResults.stream()
        .map(it -> it.getResultStatus().reverse())
        .collect(
            groupingBy(it -> it, collectingAndThen(counting(), val -> new Count(Math.toIntExact(val))))
        );

    return new DealerResult(name, collect);```

Reply:
stream 의 세계는 정말 끝이 없네요...! 

1. map 을 통해 reverse 한 Stream<ResultStatus>
2. collect groupBy 를 통해 Map 화
3. ResultStatus -> ResultStatus 를 key - Function.identity
4. counting 하여 모은후 해당 값을 통해 Count 생성
보면 바로 이해는 할 수 있으나 , 생각하기는 정말 어렵네요 😭
추가적으로 , Math.toIntExact 라는 메소드가 있는지도 몰랐네요!

해당 로직에 대해서는 다른 곳에서도 유용하게 사용 할 수 있을거 같습니다!
( 이런게 정말 위에서 웨지가 말해준 것 처럼 가독성 Bad 적인 Stream 인거 같네요 ㅋㄱㅋㄱㅋㄱㄲ )

Reply:
습관적으로 int , integer 를 사용하는데
Long 으로 선언 하는것도 나쁘지 않은 방법인거 같습니다! 
---
toMap , groupingBy  , collectingAndThen 에 대해서는 
학습할 필요가 있는거 같습니다!